community/kernel/src/main/java/org/neo4j/kernel/api/index/IndexAccessor.java

```diff
@@ -90,6 +91,17 @@
      */
     ResourceIterator<File> snapshotFiles() throws IOException;
 
+    /**
+     * Verifies that each value in this index is unique.
+     * Index is guaranteed to not change while this call executes.
+     *
+     * @param propertyAccessor {@link PropertyAccessor} for accessing properties from database storage
+     * in the event of conflicting values.
+     * @throws IndexEntryConflictException for first detected uniqueness conflict, if any.
+     * @throws IOException on error reading from source files.
+     */
+    void verifyDeferredConstraints( PropertyAccessor propertyAccessor ) throws IndexEntryConflictException, IOException;
```

Comment:
It is a little bit strange that the `verifyDeferredConstraints` always verifies that the index has unique entries, while the IndexPopulator variant only does so if the Index is actually supposed to be unique. The way these are used, that makes sense, but for understanding it would be nicer with wither consistent behaviour or different method names.